### PR TITLE
Use protobufs to communicate with trace api

### PR DESCRIPTION
**Pull request by matthewloring**
Protos were imported from:
  https://github.com/google/googleapis
  https://github.com/google/protobuf

### CODE REVIEW

**Comment by matthewloring** [comment]
@ofrobots This is passing!!!

File: third_party/proto/google/api/annotations.proto [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import "../../google/protobuf/descriptor.proto";
+
+option java_multiple_files = true;
+option java_outer_classname = "AnnotationsProto";
+option java_package = "com.google.api";
+
+extend google.protobuf.MethodOptions {
+  // See `HttpRule`.
+  HttpRule http = 72295728;
+}
```
Comment (ofrobots) [inline third_party/proto/google/api/annotations.proto]: missing new line at end

Comment (ofrobots) [inline third_party/proto/google/api/annotations.proto]: But this is a copied file. Nevermind if the original doesn't have it.

File: lib/trace.js [hunk 2]
```diff
@@ -23,8 +23,8 @@
  * @constructor
  */
 function Trace(projectId, traceId) {
-  this.projectId = projectId;
-  this.traceId = traceId;
+  this.project_id = projectId;
+  this.trace_id = traceId;
```
Comment (ofrobots) [inline lib/trace.js]: Do we even need this file anymore? Can't we construct the protobuf class directly?

File: lib/trace-span.js [hunk 3]
```diff
@@ -19,17 +19,28 @@
...
+    seconds: Math.floor(time / 1000),
+    nanos: Math.floor((time % 1000) * 1000000)
+  };
+};
+
+/**
  * Creates a trace span object.
  * @constructor
  */
 function TraceSpan(name, spanId, parentSpanId, time) {
```
Comment (ofrobots) [inline lib/trace-span.js]: Similarly, I don't think we need this class anymore.

File: third_party/proto/google/api/annotations.proto (hunk 1 above)
Comment (matthewloring) [inline third_party/proto/google/api/annotations.proto]: The original does not have it.

File: lib/trace.js (hunk 2 above)
Comment (matthewloring) [inline lib/trace.js]: We can, I'd like to land a functionality CL as this is already getting really big and then independently land a refactor that removes our structural classes.

File: lib/trace-span.js (hunk 3 above)
Comment (matthewloring) [inline lib/trace-span.js]: I agree. I can get rid of our structural classes now if we don't mind making this CL larger. Or I can do it in the next one.

Comment (ofrobots) [inline lib/trace-span.js]: That works for me.

Comment (ofrobots) [inline lib/trace-span.js]: To clarify; doing it in the next one is fine.

**Comment by ofrobots** [comment]
The changes here look good, but this should not be landed yet. The pre-requisites would be:
1. grpc release w/ the necessary support.
2. e2e testing infrastructure should be functional enough to test this.
3. performance evaluation.

**Comment by matthewloring** [comment]
After the performance review done by @cristiancavalli, it seems that grpc is not a feasible solution at this time.

**Pull request closed without merging**